common/lib/protocol-definitions/src/summary.ts

```diff
@@ -21,41 +21,88 @@ export interface ISummaryCommitter {
     date: string;
 }
 
+/**
+ *  Represents a node from the Summary Tree.
+ */
+// eslint-disable-next-line @typescript-eslint/no-namespace
 export namespace SummaryType {
     export type Tree = 1;
     export type Blob = 2;
     export type Handle = 3;
     export type Attachment = 4;
 
-    export const Tree: Tree = 1 as const;
-    export const Blob: Blob = 2 as const;
-    export const Handle: Handle = 3 as const;
-    export const Attachment: Attachment = 4 as const;
+    /**
+     *  Represents a sub-tree in the summary.
+     */
+     export const Tree: Tree = 1 as const;
+
+     /**
+      * Represents a blob of data that is added to the summary.
+      * Such as the user data that is added to the DDS or metadata added by runtime
+      * such as data store / channel attributes.
+      */
+     export const Blob: Blob = 2 as const;
+
+     /**
+      * Path to a summary tree object from the last successful summary.
+      */
+     export const Handle: Handle = 3 as const;
+
+     /**
+      * Unique identifier to larger blobs uploaded outside of the summary.
+      * Ex. DDS has large images or video that will be uploaded by the BlobManager and
+      * receive an Id that can be used in the summary.
+      */
+     export const Attachment: Attachment = 4 as const;
 }
 export type SummaryType = SummaryType.Attachment | SummaryType.Blob | SummaryType.Handle | SummaryType.Tree;
 
 export type SummaryTypeNoHandle = SummaryType.Tree | SummaryType.Blob | SummaryType.Attachment;
 
+/**
+ * Path to a summary tree object from the last successful summary indicating the summary object hasn't
+ * changed since it was uploaded.
+ * To illustrate, if a DataStore did not change since last summary, the framework runtime will use a handle for the
+ * entire DataStore instead of re-sending the entire subtree. The same concept applies for a DDS.
+ * An example of handle would be: '/<DataStoreId>/<DDSId>'.
+ * Notice that handles are optimizations from the Fluid Framework Runtime and the DDS is not aware of the handles.
+ */
 export interface ISummaryHandle {
     type: SummaryType.Handle;
 
-    // No handles, all other SummaryType are Ok
+    /**
+     * Type of Summary Handle (SummaryType.Handle is not supported).
+     */
     handleType: SummaryTypeNoHandle;
 
-    // Stored handle reference
+    /**
+     * Unique path that identifies the stored handle reference.
+     */
     handle: string;
```

Comment:
How do I get one of these?

Reply:
As a developer, I don't think you need to get it - if I got it right, it gets automatically set (id) when we create the fluid file from a summary: Ex.  createNewFluidFileFromSummary  -

Reply:
Not really. Currently, every object that wants to use this to send a path to previous summary instead of full summary has to know what the path is. For example, the summarizer node for data store context and channel contexts tracks this path and adds it.
This path is basically its path in the container's object tree. For instance, for DDS it is /<dataStoreId>/<ddsId>.

Reply:
I believe the way it should it be is similar to how summary tree is built. Each node just adds its summary tree when asked for it. It's the responsibility of the caller to add this tree against the correct path. This way by the time the summary tree reaches the root, all paths are fully built w.r.t. root.

Reply:
Let's sync up  ?

Reply:
sure

Reply:
Just as a reference- the only place I found that adds the handle to the tree is when we call encodeSummary  (builder.addHandle).

Reply:
This is the main usage of summary handle in our code - https://github.com/microsoft/FluidFramework/blob/aeac6ba5183baed5841a9741b7ac192242ab36d2/packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts#L100.

The `encodeSummary` bit is part of this feature called differential summary which we don't use anymore.

Reply:
At the summarizer node level,  if we are tracking it, it is stored under _latestSummary.fullPath.path